src/program/lwaftr/tests/propbased/genyang.lua

```diff
@@ -464,44 +477,45 @@ local types = { "int8", "int16", "int32", "int64", "uint8", "uint16",
                 "ipv6-address", "ipv6-prefix", "mac-address", "string",
                 "binary" }
 
-function generate_config_xpath_and_val(schema_name)
-   if not schema_name then
-      schema_name = choose(schemas)
-   end
+function generate_config_xpath_and_val(schema_name, generate_invalid)
+   schema_name = schema_name or choose(schemas)
    local schema = schema.load_schema_by_name(schema_name)
    local r = math.random()
    local path, val
 
    -- once in a while, generate a nonsense value
    if generate_invalid and r < 0.05 then
-     path = generate_xpath(schema, false)
-     val = value_from_type({ primitive_type=choose(types) })
+     path = generate_xpath(schema, true)
+     val = value_from_type({ primitive_type=choose(types) }, generate_invalid)
    else
-     path, val = generate_xpath_and_val(schema)
+     path, val = generate_xpath_and_val(schema, true, generate_invalid)
    end
 
    return path, val, schema_name
 end
 
-function generate_config_xpath_state(schema_name)
-   if not schema_name then
-      schema_name = choose(schemas)
-   end
-   local schema      = schema.load_schema_by_name(schema_name)
-   local path = generate_xpath(schema.body["softwire-state"], true)
-   return "/softwire-state" .. path, schema_name
+function generate_state_xpath(schema_name, generate_invalid)
+   schema_name = schema_name or choose(schemas)
+   local schema = schema.load_schema_by_name(schema_name)
+   return generate_xpath(schema, false, generate_invalid), schema_name
 end
 
 function selftest()
+   print('selftest: program.lwaftr.tests.propbased.genyang')
    local data = require("lib.yang.data")
    local path = require("lib.yang.path")
-   local schema = schema.load_schema_by_name("snabb-softwire-v1")
+   local schema = schema.load_schema_by_name("snabb-softwire-v2")
    local grammar = data.config_grammar_from_schema(schema)
 
-   path.convert_path(grammar, generate_xpath(schema))
+   for i=1,1000 do
+      local subpath, value = generate_xpath_and_val(schema, true)
+      -- print(i, subpath, value)
```

Comment:
Doesn't matter too much but maybe these print debugs can be removed?

Reply:
Done!